backend/cmm_helpers.ml

```diff
@@ -2526,172 +2566,202 @@ let tuplify_function arity =
 
 let max_arity_optimized = 15
 
-let final_curry_function ~nlocal ~arity =
+let machtype_stored_size t =
```

Comment:
Is it worth having this function now that 32-bit backend support has been removed?

Reply:
The point of this function is to prepare support for complex machtypes such as `[|Float; Int|]` in the future, so this function is still needed. However, with 32-bit backend support removed, we could probably replace its body with `Array.length t` (skipping the checks for `Addr`), but I still think having the function would be better for code readability.

Reply:
Yeah ok, changing it as you suggest seems fine

Reply:
Changed to simply compute the array length in case `Arch.size_int` is 8, but I didn't completely remove the support for it since the function below handled this case as well (I can remove the support in both however, if you prefer).

Reply:
This seems ok